src/graphapi/find_engine.cpp

```diff
@@ -0,0 +1,341 @@
+    }
+
+public:
+    static std::unique_ptr<GraphPatternMatcher> Make()
+    {
+        return std::make_unique<MHA_Fwd_F8_Pattern>();
+    }
+
+    std::string_view name() const final
+    {
+        static const char* n = "mha_fwd_f8";
+        return n;
+    }
+
+    bool matches(const OpGraph* graph_ptr) const final
+    {
+        assert(graph_ptr);
+        return isIsomorphic(*graph_ptr, getPatternGraph());
+    }
+
+    std::vector<Engine> getEngines(OpGraph* graph_ptr) const override
+    {
+
+        assert(graph_ptr);
+        assert(matches(graph_ptr));
+        auto& graph = *graph_ptr;
+
+        miopenProblem_t mha_prob;
+        MhaDescriptor mha_desc;
+
+        float attn_scale                          = std::numeric_limits<float>::quiet_NaN();
+        std::shared_ptr<TensorInfoMap> tensor_map = extractFind20Tensors(graph, &attn_scale);
+        assert(attn_scale != std::numeric_limits<float>::quiet_NaN());
+
+        mha_desc.SetParams(attn_scale);
+
+        auto s = miopenCreateMhaProblem(&mha_prob, &mha_desc, miopenProblemDirectionForward);
+        MIOPEN_THROW_IF(s != miopenStatusSuccess, "failed while creating problem for mha fwd");
+
+        for(auto& [k, v] : *tensor_map)
+        {
+            s = miopenSetProblemTensorDescriptor(mha_prob, v.mEnumId, &(v.mTensDesc));
+            MIOPEN_THROW_IF(s != miopenStatusSuccess,
+                            "failed while setting tensor descriptor for mha fwd");
+        }
+
+        std::vector<miopenSolution_t> solutions(10);
+        size_t num_found = 0;
+        s                = miopenFindSolutions(
+            graph.getHandle(), mha_prob, nullptr, solutions.data(), &num_found, solutions.size());
+        MIOPEN_THROW_IF(s != miopenStatusSuccess, "failed while finding solutions for mha fwd");
+
+        solutions.resize(num_found);
+
+        std::vector<Engine> engines;
+
+        size_t i = 0;
+        for(const auto& sol : solutions)
+        {
+            std::shared_ptr<GraphPatternExecutor> exec = GraphExecutorFind20::make(sol, tensor_map);
+
+            engines.emplace_back(
+                EngineBuilder().setGraph(graph_ptr).setExecutor(exec).setGlobalIndex(i).build());
+            ++i;
+        }
+
+        return engines;
+    }
+};
+
+/*
+class FwdConvResAddBiasActPattern : public GraphPattern
+{
+public:
+    static std::unique_ptr<GraphPattern> Make()
+    {
+        return std::make_unique<FwdConvResAddBiasActPattern>();
+    }
+};
+*/
+
+std::vector<Engine> findEngines(OpGraph* graph)
+{
+    assert(graph);
+
+    std::vector<std::unique_ptr<GraphPatternMatcher>> patterns;
+    patterns.emplace_back(MHA_Fwd_F8_Pattern::Make());
+
+    for(const auto& p : patterns)
+    {
+        if(p->matches(graph))
+        {
+            MIOPEN_LOG_I("Matched against pattern: " << p->name());
+            return p->getEngines(graph);
+        }
+    }
+
+    MIOPEN_LOG_E("No patterns matched against the graph");
+    MIOPEN_THROW("Solution not found");
```

Comment:
[R] Output of MIOPEN_LOG_E won't be visible anyway (on most Linux systems at least)
```suggestion
    MIOPEN_THROW("No patterns matched against the graph. Solution not found");
```